src/autoscaler/scalingengine/server/applock_test.go

```diff
@@ -0,0 +1,106 @@
+package server_test
+
+import (
+	. "autoscaler/scalingengine/server"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"time"
+)
+
+var (
+	ch      chan string
+	appLock *AppLock
+	err     interface{}
+)
+
+var _ = Describe("Applock", func() {
+	BeforeEach(func() {
+		appLock = NewAppLock()
```

Comment:
Ok, so now taking a closer look, what you did conceptually is correct but not quite the right implementation. You are layering a blocking request within something that expects to be periodic.

hasAcquiredLock should just be 
```
select {
  case <-ch:
    return true
  default:
    return false
}
```
Time shouldn't matter if you are using Eventually/Consistently around it.